### PR TITLE
libsel4debug: use proper closing bracket

### PR DESCRIPTION
**Pull request by axel-h**
use proper closing bracket in the debug output

### CODE REVIEW

File: libsel4debug/src/bootinfo.c [hunk 1]
```diff
@@ -18,11 +18,11 @@ void debug_print_bootinfo(seL4_BootInfo *info)
...
-    printf("Empty slots:     [%lu --> %lu)\n", (long)info->empty.start, (long)info->empty.end);
-    printf("sharedFrames:    [%lu --> %lu)\n", (long)info->sharedFrames.start, (long)info->sharedFrames.end);
-    printf("userImageFrames: [%lu --> %lu)\n", (long)info->userImageFrames.start, (long)info->userImageFrames.end);
-    printf("userImagePaging: [%lu --> %lu)\n", (long)info->userImagePaging.start, (long)info->userImagePaging.end);
-    printf("untypeds:        [%lu --> %lu)\n", (long)info->untyped.start, (long)info->untyped.end);
+    printf("Empty slots:     [%lu --> %lu]\n", (long)info->empty.start, (long)info->empty.end);
+    printf("sharedFrames:    [%lu --> %lu]\n", (long)info->sharedFrames.start, (long)info->sharedFrames.end);
+    printf("userImageFrames: [%lu --> %lu]\n", (long)info->userImageFrames.start, (long)info->userImageFrames.end);
+    printf("userImagePaging: [%lu --> %lu]\n", (long)info->userImagePaging.start, (long)info->userImagePaging.end);
+    printf("untypeds:        [%lu --> %lu]\n", (long)info->untyped.start, (long)info->untyped.end);
```
Comment (lsf37) [inline libsel4debug/src/bootinfo.c:25]: The `)` means the end is not included in the interval. Is it?

**Comment by kent-mcleod** [comment]
> use proper closing bracket in the debug output

The closing bracket is already correct.  The proposed change would mean it no longer matches with the opening bracket.  The output isn't trying to be mathematical notation. 

I don't think there is a good enough motivation for this change. This output has already been used for several years and there's some value in it maintaining a consistent syntax across multiple versions.

**Comment by kent-mcleod** [comment]
> @kent-mcleod: I'm confused now. The opening bracket it `[` and the closing bracket is `)`. My proposal is changing the closing bracket to `]` to match the opening bracket.

Sorry, I don't know what I was thinking here.  @lsf37 points out that the notation is currently following mathematical range notation to communicate that the end values are exclusive not inclusive (https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints) which I believe means that the brackets were intentionally mismatched.

**Comment by pingerino** [comment]
Yes, the brackets were intentionally mismatched.

**Pull request closed without merging**